[STN-62] Scaffold templates/patterns for HumSci Base

nrambeck opened the pull request:
# READY FOR REVIEW

# Summary

- Added the Drupal templates and UI Patterns from the legacy HumSci theme to support all current features.
- Also added configuration for Colorful theme block placement and configuration.
- re-organized the patterns folder to be a flat structure (no more atoms, molecules, organisms)
- Added more Drupal templates from Stanford Basic (blocks, menus, page title, html, node, page, region)
- Added favicon and screenshot (for Themes/Appearance UI)
- Removed Highlighted Region from theme (based on how the Stanford Basic theme is setup)
- Removed Google Analytics tag manager

# Steps to Test

If using Lando, prefix drush commands with `lando`.

1. Import configuration (`drush cim`) 
1. Clear caches (`drush cr`)
1. Enable the Colorful theme (`drush theme:enable humsci_colorful`)
1. Ensure various pages render without any errors
1. Compare Blocks placement on the Colorful theme to that of the legacy HumSci theme. Ensure they match.
1. Check `/patterns` page (that it works, but appears unstyled)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

Commit:
STN-62 Add patterns, templates and favicon from HumSci Theme as a starting point

Also include a screenshot file for the theme

Commit:
STN-62 Add configuration for Block placement and configuration for the HumSci Basic theme

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

silverli commented on docroot/themes/humsci/humsci_basic/humsci_basic.info.yml:
```diff
@@ -15,3 +15,8 @@ regions:
   page_bottom: 'Page bottom'
+
+component-libraries:
+  humsci_basic:
```
looks like .gitkeep files got added in this commit FYI

nrambeck replied:
I restructured this entire folder so we no longer having empty folders like this.

bryanbraun commented:
To follow up on:

- Are there any generic templates from stanford_basic that we can include here as well?
- Can we bring over the favicon now?
- Theme preview image?
- Do we still need to use Atoms / Molecules / Organisms?

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

nrambeck commented:
I made the following changes based on feedback:
- re-organized the patterns folder to be a flat structure (no more atoms, molecules, organisms)
- Added more Drupal templates from Stanford Basic (blocks, menus, page title, html, node, page, region)
- Added favicon and screenshot (for Themes/Appearance UI)
- Removed Highlighted Region from theme (based on how the Stanford Basic theme is setup)

silverli commented on docroot/themes/humsci/humsci_basic/templates/block/block--system-menu-block--main.html.twig:
```diff
@@ -0,0 +1,37 @@
+{#
+/**
+ * @file
```
why do we have system-menu-block--main and system-menu-block below?

nrambeck replied:
The `--main` one represents the main content block only, which has no title in the block itself.

silverli commented on docroot/themes/humsci/humsci_basic/templates/contrib/patterns-variant-meta-information.html.twig:
```diff
@@ -0,0 +1,27 @@
+{#
+/**
+ * @file
+ * UI Pattern variant meta information.
```
where did this file come from?

silverli replied:
this was in stanford basic. check for /patterns page when running locally

silverli commented on docroot/themes/humsci/humsci_basic/templates/menus/macros/nav-menu.twig:
```diff
@@ -0,0 +1,37 @@
+{#
+/**
+ * Macro for creating nested menus.
```
where did this file come from as well? did you custom make it?

silverli replied:
it's from the latest stanford basic theme (4.x)

silverli commented on docroot/themes/humsci/humsci_basic/templates/page.html.twig:
```diff
@@ -1,45 +1,27 @@
-<header id="header">
```
so we switched this one out from the su_humsci theme one to the stanford_basic yes?

nrambeck replied:
Ending up using the latest Stanford Basic version of this template.

silverli commented on config/default/block.block.humsci_basic_messages.yml:
```diff
@@ -10,7 +10,7 @@ _core:
   default_config_hash: IN43gws2bI68lCkbtZPAxNJrQl7KqQJt2nhzkuJAqoM
 id: humsci_basic_messages
 theme: humsci_basic
-region: highlighted
```
what is the reason for removing this?

silverli replied:
this region didn't exist in stanford basic, and it lived with help in su_humci

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

nrambeck commented:
I added Masonry patterns back in and removed Google Tag Manage JS from the page template.

silverli reviewed (changes requested):
I think this is just about ready, but I just have a couple further clarification questions I left in the comments that I'd like to go over Monday before merge.

silverli reviewed (approved):
This looks great. Here are screenshots of the new theme working, just a bit unstyled on the home page and patterns page:

<img width="498" alt="Screen Shot 2019-12-16 at 11 21 39 AM" src="https://user-images.githubusercontent.com/12848168/70924025-de498f00-1ff6-11ea-8c4b-8845e854a73b.png">
<img width="495" alt="Screen Shot 2019-12-16 at 11 22 03 AM" src="https://user-images.githubusercontent.com/12848168/70924032-e1447f80-1ff6-11ea-8e37-7659f5ad9e5d.png">
